### ISSUE TITLE
Inconsistent types for line numbers

### ISSUE THREAD

**Issue by variar** [variar/klogg#13]
Although reading from original file produces int64 line numbers, int32 line numbers are used in UI. I'm not sure all narrowing conversions are valid. Also right now it is impossible to have more than UINT32_MAX matches.

Might need to switch to size_t for all line numbers

**Comment by variar** [comment]
Opening files with more than UINT32_MAX is broken in klogg as well as upstream. Index builds fine, but line number counter overflows.

**Comment by variar** [comment]
Klogg uses `QString::length` and `QTextCodec::toUnicode` to handle individual strings. So there will be no support for files that have lines with more than 2147483647 characters.

I've decided to keep original limit for total number of lines in file. Files with more than 4294967295 lines will not be displayed correctly. 

Klogg uses QIODevice interface to read data from filesystem. This implies that maximum supported file size is 9223372036854775807 bytes (~9223 petabytes). It seems very unlikely to cause any trouble.

I've used stong typing technique from [Fluent C++](https://www.fluentcpp.com/2016/12/08/strong-types-for-strong-interfaces/) to express line related types in code (line number, line length, offset and count). This prevents from accidentally passing wrong type (for example, line number instead of number of lines) and allows to limit operations on that types (adding line numbers is not allowed, subtracting line numbers produces a number of lines, but line number can be incremented/decremented etc.). 

I hope more old bugs were fixed during this refactoring that introduced.
